src/MicroplateSet/MicroplateSet.php

```diff
@@ -50,4 +50,18 @@ public function locationFromPosition(int $setPosition, FlowDirection $direction)
             $this->plateIDs()[$plateIndex]
         );
     }
+
+    /**
+     * @param Location<TCoordinateSystem> $location
+     */
+    public function positionFromLocation(Location $location, FlowDirection $direction): int
+    {
+        $positionOnPlate = $location->coordinates
+            ->position($direction);
+
+        $positionOnSet = $positionOnPlate * array_keys($this->plateIDs())[$location->plateID];
+        assert(is_int($positionOnSet));
```

Comment:
This makes no sense to me, `array_keys($this->plateIDs())` will be something like `[0, 1, ..]` - then you index into this with another `plateID`? Maybe you need `array_search`?